Domains: Update domain item list layout

leonardost opened the pull request:
#### Changes proposed in this Pull Request

This PR corresponds to item 3 of pcYYhz-e4-p2, which is a redesign of the domain items list in the site's domains page.

The main updates are:
- Removal of the primary domain card, which is replaced by a "Primary site address" green badge
- The domain item list header was simplified and an "Add a domain to this site" dropdown button was added
- Each domain item component was simplified, with auto-renewal and email information shown as text and the domain connection status shown at the bottom of each card

### Screenshots

This is how the site domains management page looked like:

<img width="1053" alt="Screen Shot 2021-07-22 at 11 29 11" src="https://user-images.githubusercontent.com/5324818/126660739-c2db5f68-56f2-4fd1-bae5-0c207aa2e833.png">

And how it should look like after the update:

<img width="1064" alt="Screen Shot 2021-08-03 at 13 27 53" src="https://user-images.githubusercontent.com/5324818/128052287-19ee4d03-21a3-43ee-bb2a-9194d6ff686c.png">

When the "Add a domain to this site" dropdown button is clicked:

<img width="295" alt="Screen Shot 2021-08-03 at 13 28 24" src="https://user-images.githubusercontent.com/5324818/128052343-1ac8a02f-8cc2-497e-ae60-0ed5360a37f7.png">

Detail of domain item "Verifying connection" status:

<img width="1055" alt="Screen Shot 2021-08-03 at 13 30 24" src="https://user-images.githubusercontent.com/5324818/128052364-c95b31ef-4510-47ac-a0ec-2b052d9ea7a5.png">

And when viewing on a mobile device:

<img width="418" alt="Screen Shot 2021-08-03 at 13 29 02" src="https://user-images.githubusercontent.com/5324818/128052390-e4f992de-d14e-4c82-bc65-222f7dcb9839.png">

#### Testing instructions

The ideal way to test this would be to check the most domain statuses as you can and ensure the layout and info shown are accurate:
- Mapped domains
  - Mapped less than 72 hours ago and still not resolving to WPCOM (should show "Verifying connection" status)
  - Mapped more than 72 hours ago and still not resolving to WPCOM (should show "Connection error" status)
- Registered domains
- Transferred domains

With:
- Configured emails
- Expiring soon (in 30 days or less)
- Any other variations you can think of

Also please check the "Manage all domains" page to ensure information is shown correctly there as well, as they use the same `DomainItem` component.

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

leonardost committed:
Comment primary domain card

leonardost committed:
Rename "Mapped Domain" to "Connected Domain"

leonardost committed:
Increase domain name size and remove "domain-item__primary" CSS class

leonardost committed:
Add auto-renew status besides site meta information

leonardost committed:
Add email status besides site meta and auto-renew information

leonardost committed:
Decrease space between items in site meta information

a8ci18n commented on client/my-sites/domains/domain-management/list/domain-item.jsx:
```diff
@@ -401,9 +401,97 @@ class DomainItem extends PureComponent {
+		return (
+			<span className="domain-item__meta-item">
+				<Button compact onClick={ this.addEmailClick } className="domain-item__email-button">
+					{ translate( 'Add email', { context: 'Button label' } ) }
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **45** times:
`translate( 'Add email' )` *ES Score: 11*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/domain-item-list-updates)


:information_source: This string already exists without a **context**. Only add a context if the meaning of the string is very specific.

leonardost committed:
Add primary site address badge placeholder

leonardost committed:
Remove DomainNotice component from besides the domain name

leonardost committed:
Show placeholder in place of site meta when details are not loaded yet

leonardost committed:
Remove action column items

leonardost committed:
Move renderAutoRenew method back to its original position

leonardost committed:
Move renderEmail method back to its original position

leonardost committed:
Revert unwanted changes

a8ci18n commented on client/my-sites/domains/domain-management/list/domain-item.jsx:
```diff
@@ -306,56 +294,62 @@ class DomainItem extends PureComponent {
-			</Button>
+			<span className="domain-item__meta-item">
+				<Button compact onClick={ this.addEmailClick } className="domain-item__email-button">
+					{ translate( 'Add email', { context: 'Button label' } ) }
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **45** times:
`translate( 'Add email' )` *ES Score: 11*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/domain-item-list-updates)


:information_source: This string already exists without a **context**. Only add a context if the meaning of the string is very specific.

leonardost committed:
Add proper primary site address badge

leonardost committed:
Update domain items list header

leonardost committed:
Refactor inline style into CSS class

leonardost committed:
Remove unused domain items list header code and CSS styles

leonardost committed:
Replace "COnnected domain" by "Managed by external provider"

leonardost committed:
Add "verifying" domain status styling for domain items

leonardost committed:
Show domain connection status notices at the bottom of the domain item cards

leonardost commented on client/my-sites/domains/domain-management/components/domain-notice/style.scss:
```diff
@@ -9,14 +9,17 @@
-	color: var( --color-success-50 );
+.domain-notice__success,
+.domain-notice__verifying {
+	svg {
+		color: var( --color-success-50 );
+	}
 }
 
-.domain-notice__warning {
+.domain-notice__warning svg {
 	color: var( --color-warning-40 );
 }
 
-.domain-notice__alert {
+.domain-notice__alert svg {
```
Now only the icon will be colored.

leonardost committed:
Only show domain connection status message if there is any to show

leonardost committed:
Add domain connection error status for domain items

a8ci18n commented on client/lib/domains/resolve-domain-status.js:
```diff
@@ -45,12 +53,45 @@ export function resolveDomainStatus(
+					}
+				);
+				return {
+					statusText: translate( 'Connection error' ),
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **21** times:
`translate( 'Publicize Connection Error' )` *ES Score: 7*

leonardost committed:
Add domain connection error verification

leonardost committed:
Remove primary domain card component

leonardost committed:
Hide primary site address badge from domain when in the manage all domains page

leonardost committed:
Hide "Add domain to this site" button when in the manage all domains page

leonardost committed:
Make new strings translatable

leonardost committed:
Remove comment

a8ci18n commented on client/lib/domains/resolve-domain-status.js:
```diff
@@ -44,13 +52,45 @@ export function resolveDomainStatus(
 				return {
-					statusText: translate( 'Complete setup' ),
-					statusClass: 'status-warning',
+					statusText: translate( 'Connection error' ),
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **21** times:
`translate( 'Publicize Connection Error' )` *ES Score: 7*

leonardost committed:
Remove unit tests that tested the primary domain card which was removed

leonardost commented on client/my-sites/domains/domain-management/list/test/index.js:
```diff
@@ -117,129 +117,6 @@ describe( 'index', () => {
-			} );
-		} );
-	} );
-
```
This test set tested the primary domain card and `set primary domain mode` that were removed.

leonardost committed:
Show domain item list header in small screens

leonardost committed:
Tweak CSS to improve layout in small screens

leonardost committed:
Make more CSS tweaks for mobile

delputnam commented on client/my-sites/domains/domain-management/list/add-domain-button.jsx:
```diff
@@ -17,11 +17,16 @@ import PopoverMenuItem from 'calypso/components/popover/menu-item';
 class AddDomainButton extends React.Component {
 	static propTypes = {
 		selectedSiteSlug: PropTypes.string,
+		specificSiteActions: PropTypes.bool,
```
It's nice that you handled this case, @leonardost ! 🙇 
Since this will be unnecessary once we merge #54635, could you please add a task for us to clean up the unused part of the code that won't be necessary once that PR is merged? I just don't want to forget to clean it up. 😅

leonardost replied:
Sure! I created it here 1146722293412575-as-1200645972957212.

delputnam commented on client/my-sites/domains/domain-management/components/domain-notice/index.jsx:
```diff
@@ -14,7 +14,7 @@ import './style.scss';
 export default class DomainNotice extends React.Component {
 	static propTypes = {
 		text: PropTypes.string,
```
I know this was already like this, but just noticed it while testing. Text that includes react elements is passed in as an array, so I guess we should allow that here too.
```suggestion
		text: PropTypes.oneOfType( [ PropTypes.string, PropTypes.array ] ),
```

leonardost replied:
Good catch, I saw the warnings in the console but forgot to fix this 🙇

delputnam commented on client/my-sites/domains/domain-management/list/domain-item.jsx:
```diff
@@ -192,36 +194,32 @@ class DomainItem extends PureComponent {
-	renderTransferLock() {
-		const { domainDetails } = this.props;
-		if ( domainDetails?.type === domainTypes.REGISTERED ) {
-			return this.renderCheckmark( domainDetails?.isLocked );
```
Since we don't use `renderCheckmark()` any longer, we can remove this method.

leonardost replied:
👍

delputnam reviewed (approved):
Reviewed, tested and this LGTM. Only a couple of minor comments.

delputnam commented:
@leonardost Could you also please have a look at the failing tests and double-check that those aren't due to a side-effect of the tests we're removing in this PR.

leonardost committed:
Update PropTypes of text attribute in DomainNotice

Was only `PropTypes.string`, now it can be `PropTypes.string` or
`PropTypes.array`.

leonardost committed:
Remove unused methods from DomainItem

leonardost committed:
Update AddDomainButton className when used for specific site actions

This was done to check if E2E tests work again.

leonardost commented:
If you received a random request to review this, I'm sorry, I didn't mean to 😅

leonardost committed:
Merge branch 'trunk' into update/domain-item-list-updates

leonardost committed:
Revert changes done in 1609d078ba to try to fix E2E tests

delputnam reviewed (approved):
Changes LGTM.  Thanks for making them. 👍

poligilad-auto commented:
Looking really cool @leonardost!

Some small tweaks:
- _.domain-item__title_ font-weight should be 500. 
- please add 4px spacing between the two divs in _.list__domain-link_.
- the _add email_ link should be styled the same as the other meta items, so 14px as well.
- _add a domain to this site_ button - can we have the popover menu align to whole button and not just the _arrow_ icon?

leonardost committed:
Adjust spacings according to design review

leonardost committed:
Fix email button-link size and alignment

leonardost committed:
Position popover menu in the middle of the button

leonardost commented:
Thanks for the review @delputnam and @poligilad-auto!

@poligilad-auto I've updated the code with your observations, could you take a look at it again to see if it's ok?

Also, I had forgotten to mention, in your original Figma design you used a different icon for the "Verifying connection" status:
<img width="404" alt="Screen Shot 2021-07-27 at 18 02 23" src="https://user-images.githubusercontent.com/5324818/127227272-fb6e3009-158e-4490-950f-53be6a000355.png">
I used this one instead - the closest one I could find in the [GridIcons set](http://automattic.github.io/gridicons/) (the official Calypso icon set) - is that ok?
<img width="353" alt="Screen Shot 2021-07-27 at 18 04 29" src="https://user-images.githubusercontent.com/5324818/127227724-2fced129-30b6-41d2-a74e-01248f939a0f.png">

Edit: And I just noticed the bold text should be green too 😆

leonardost committed:
Change color of bold text in domain connection status text

hambai commented:
> I used this one instead - the closest one I could find in the [GridIcons set](http://automattic.github.io/gridicons/) (the official Calypso icon set) - is that ok?

Just to mention that the icon from the Figma file is `compare_arrows` from the [material icons set](https://fonts.google.com/icons?selected=Material+Icons). It is currently not added in the `material-design-icons` package but we can easily add it if needed

poligilad-auto commented:
Hey @leonardost. Your changes look great.
Thanks for noticing that bit, sorry I missed it somehow.. I also noticed a few other small things.

1. Re icons: I was using [these icons](https://www.figma.com/file/e4tLacmlPuZV47l7901FEs/WordPress-Design-Library?node-id=1%3A2) in my designs which are the Gutenberg ones. Is it possible to use them or do we have to use the Gridicons?
2. The font color for _verifying connection_ should be var(--color-success-60), the color for _connection error_ should be var(--color-error).
3. The title _Your site domains_ should be $font-body-small.
4. I noticed that the ellipsis icon on the right isn't visually aligned to the edge of the right side. It would be great if we could align it to the end, just like the _add a domain to this site_ button above.

xpurichan commented:
I was confused at why we have both the "Add a domain" and "Add a domain to this site" buttons here, it seems very redundant, are there other plans to further change this setup?

Should there be a warning/notice when a domain is registered to us and pointing to another host? I know we chatted about this in our most recent meeting but I can't recall what was decided. I understand that a customer might intentionally register a domain with us and then point it elsewhere, but it's a problem when a customer has transferred-in a domain. We do have a notice about this, but it's only visible after clicking on the domain from this Domains page.

Focusing back on the task at hand: I think we do need some margin added to the "Add a domain" button for mobile. Right now it's hitting the edge of the screen (shown here in Chrome using an iPhone X simulation):

![Screen Shot 2021-07-28 at 11 45 00 AM](https://user-images.githubusercontent.com/33190975/127362969-047bb5a6-0d98-41ba-b715-6d9b447c5ab5.jpg)

leonardost committed:
Change "Verifying connection" texto color

leonardost committed:
Update "Your site domains" size and align ellipsis menu with right margin

leonardost committed:
Use Material Design's compare_arrows icon in verifying connection status

leonardost commented:
@delputnam  @hambai @poligilad-auto @xpurichan Thank you all for the review!

@poligilad-auto Thanks for sharing that link, I'm still not familiar with all the design resources we have. Yep, we can use these icons, I've added that one to the project. I've also done the last adjustments, could you please take another look?

@xpurichan,
> why we have both the "Add a domain" and "Add a domain to this site" buttons

> I think we do need some margin added to the "Add a domain" button for mobile

Updates to the "Add a domain" button are being made in PR #54635, that's why it looks weird here 🙂 we chose to split these in smaller tasks to do them in parallel. When both PRs are finished it will look better.

> Should there be a warning/notice when a domain is registered to us and pointing to another host?

I've got to check this one, I'm not sure either 🤔

leonardost committed:
Merge branch 'trunk' into update/domain-item-list-updates

poligilad-auto commented:
@leonardost changes look good! Thanks.

I can't test the _Verifying connection_ or _Connection error_ states, but from the screen shots on **mobile** it seems that we might need more spacing between the meta line and the status line below it. I think maybe it's different from desktop? It shouldn't be.

leonardost committed:
Merge branch 'trunk' into update/domain-item-list-updates

leonardost committed:
Increase margin below domain meta status text and fix meta info in mobile view

leonardost committed:
Remove CSS style added by mistake

leonardost commented:
Hi @poligilad-auto, I increased the margin between the meta and status lines and updated the screenshots. Could you please take one more look?

poligilad-auto commented:
> Hi @poligilad-auto, I increased the margin between the meta and status lines and updated the screenshots. Could you please take one more look?

Thanks @leonardost. The spacing looks fine, it should be 16px between them and it seems like it. 
But I noticed two other things:
1. The font size for the status line should be $font-body-small. (sorry for not noticing this before)
2. The icon should be centered to the first line of the status, currently the text is a bit low. I think this changed from the last version.

leonardost committed:
Align icons vertically in domain status text

leonardost committed:
Add spaces around CSS value

a8ci18n commented on client/my-sites/domains/domain-management/list/add-domain-button.jsx:
```diff
@@ -49,35 +54,64 @@ class AddDomainButton extends React.Component {
 		const { translate } = this.props;
 
+		const label = this.props.specificSiteActions
+			? translate( 'Add a domain to this site' )
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **23** times:
`translate( 'Add a domain to a new site' )` *ES Score: 8*

leonardost committed:
Increase size of domain status text

leonardost commented:
Hi @poligilad-auto! I've vertically centered the icon in the status line, it was indeed a regression. Sorry about that.
I've also increased the size of the status text to `$font-body-small` and will update the screenshots accordingly.

leonardost committed:
Merge branch 'trunk' into update/domain-item-list-updates

poligilad-auto commented:
Thanks @leonardost.
I just noticed 2 other small things, I appologize for not noticing all these things from the beginning. 

1. On mobile can we add 4px between the "Primary site address" badge and the rest of the meta info?
2. The title _Your site domains_, the top & bottom padding should be 11px. I want to align this to the rest of the site, check out how it looks on _Posts_.

leonardost committed:
Add margin between "primary address" badge and rest of domain meta information

leonardost committed:
Replace custom Card with component SectionHeader

leonardost commented:
Hey @poligilad-auto, I've addressed these last updates 🙂 Would love if you could take another look at this.

poligilad-auto reviewed (approved):
Looking good @leonardost.

Only wanted to make sure:
We are hiding the domain item list when the user has only the free WordPress.com address, right? Because I saw the header for the list on one site where I only have the free address.

leonardost committed:
Hide site domains list header if there are no domains in the site

leonardost committed:
Hide "Manage all domains" card

leonardost committed:
Merge branch 'trunk' into update/domain-item-list-updates

leonardost committed:
Add margin between "Claim domain" card and domain list when they appear together

leonardost commented:
@poligilad-auto that was an oversight on my part, thanks for noting! Now we are hiding the domain list header when the site only has the free WPCOM address 🙂

leonardost committed:
Fix E2E tests

leonardost committed:
Merge branch 'trunk' into update/domain-item-list-updates

leonardost committed:
Add "Manage all your domains" card again"

Prevents users from having no way of accessing hte "Manage all domains"
page while the other PRs in the domains redesign are being deployed.

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/6384269

Thank you @leonardost for including a screenshot in the description! This is really helpful for our translators.

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/6384269).
